api/src/test/java/org/pmiops/workbench/db/model/StorageEnumsTest.java

```diff
@@ -1,42 +1,117 @@
-          enumToShort = m;
-        }
+  // e.g. public static Short reviewStatusToStorage(ReviewStatus s)
+  final Map<Class, Method> enumClassToStorageMethod =
+      methods.stream()
+          // domainToDomainId is stringly typed - test with test_domainId
+          .filter(m -> !m.getName().equals("domainToDomainId"))
+          .filter(m -> enumClasses.contains(firstMethodParameterType(m)))
+          .collect(Collectors.toMap(this::firstMethodParameterType, m -> m));
 
-        if (m.getReturnType().equals(enumClass)) {
-          shortToEnum = m;
-        }
-      }
+  // e.g. public static ReviewStatus reviewStatusFromStorage(Short s)
+  final Map<Type, Method> storageMethodToEnumClass =
+      methods.stream()
+          // domainIdToDomain is stringly typed - test with test_domainId
+          .filter(m -> !m.getName().equals("domainIdToDomain"))
+          .filter(m -> enumClasses.contains(m.getReturnType()))
+          .collect(Collectors.toMap(Method::getReturnType, m -> m));
 
-      for (Object e : enumClass.getEnumConstants()) {
-        Short shortVal = (Short) enumToShort.invoke(null, e);
+  // special-case test for domain ID round trip
+
+  @Test
+  public void test_domainId() {
+    for (Domain v : Domain.values()) {
+      String storageValue = CommonStorageEnums.domainToDomainId(v);
+      assertWithMessage("unmapped enum value: " + v).that(storageValue).isNotNull();
+      assertThat(v).isEqualTo(CommonStorageEnums.domainIdToDomain(storageValue));
+    }
+  }
 
-        assertThat(shortVal).named(enumClass.getName() + ":" + e.toString()).isNotNull();
-        assertThat(shortToEnum.invoke(null, shortVal)).isEqualTo(e);
+  @Test
+  public void test_noMissingMapEntries() throws InvocationTargetException, IllegalAccessException {
+    for (final Class enumClass : enumClasses) {
+      for (final Object enumValue : enumClass.getEnumConstants()) {
+        Short shortValue = enumToStorage(enumValue, enumClass);
+        assertThat(shortValue).named(enumClass.getName() + ":" + enumValue.toString()).isNotNull();
+        assertThat(storageToEnum(shortValue, enumClass)).isEqualTo(enumValue);
       }
     }
   }
+
+  private Short enumToStorage(Object enumValue, Class enumClass)
+      throws InvocationTargetException, IllegalAccessException {
+    final Object returnValue =
+        enumClassToStorageMethod.get(enumClass).invoke(INDICATES_STATIC_METHOD, enumValue);
+    assertThat(returnValue instanceof Short).isTrue();
+    return (Short) returnValue;
+  }
+
+  private Object storageToEnum(Short shortValue, Class enumClass)
+      throws InvocationTargetException, IllegalAccessException {
+    return storageMethodToEnumClass.get(enumClass).invoke(INDICATES_STATIC_METHOD, shortValue);
+  }
+
+  /**
+   * Retrieve all Enum classes in DbStorageEnums and CommonStorageEnums
+   *
+   * <p>our convention for BiMaps in DbStorageEnums/CommonStorageEnums is <Enum, Short> so we
+   * retrieve the first parameterized type
```

Comment:
nit: you might (later) also look for bimaps that don't have these properties and make noise, but I think it's ok.